### PR TITLE
Include joining functionality

### PR DESCRIPTION
**Pull request by AlexandraImbrisca**
Continuation of #15 

The main scope of this pull request is to connect the front-end with the back-end functionalities of joining groups/events. 
- The groups section includes two different containers: the first one with joined groups and the second one with all the other groups already created that the user didn't join yet.
- The group element contains a join button if the user is not a member of it yet or a show events button otherwise. The show events button will display all the events associated with that group that the user didn't join yet.
- A button is associated with each event from the list mentioned before.
- Once the user has joined an event, it will be removed from the list of events not joined.
- Each event joined can be viewed from the calendar and has a button that will open the chat room.
- The calendar will automatically display the events for the current day.

### CODE REVIEW

**Review by tatiana-s — APPROVED** [review]

File: src/main/java/com/google/lecturechat/data/Event.java [hunk 1]
```diff
@@ -95,4 +95,13 @@ public static Event createEventFromEntity(Entity eventEntity) {
...
     }
   }
+
+  @Override
+  public boolean equals(Object anotherObject) {
+    if (!(anotherObject instanceof Event)) {
+      return false;
+    }
+    Event anotherEvent = (Event) anotherObject;
+    return (this.getId() == anotherEvent.getId());
```
Comment (nataliakrylova) [inline src/main/java/com/google/lecturechat/data/Event.java:105]: (opt)  ((Event) anotherObject).getId() I think works as well

File: src/main/java/com/google/lecturechat/servlets/GroupEventsServlet.java [hunk 2]
```diff
@@ -44,13 +44,15 @@ public void init() {
...
+    Optional<String> userId = AuthStatus.getUserId(request);
+
+    if (!userId.isPresent()) {
       return;
     }
 
     try {
       long groupId = Long.parseLong(request.getParameter(GROUP_ID_PARAMETER));
-      List<Event> events = datastore.getAllEventsFromGroup(groupId);
+      List<Event> events = datastore.getAllNotJoinedEventsFromGroup(groupId, userId.get());
```
Comment (nataliakrylova) [inline src/main/java/com/google/lecturechat/servlets/GroupEventsServlet.java:55]: Please update the class documentation.

File: src/main/java/com/google/lecturechat/servlets/GroupsServlet.java [hunk 3]
```diff
@@ -42,11 +43,13 @@ public void init() {
...
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    if (!AuthStatus.isSignedIn(request)) {
+    Optional<String> userId = AuthStatus.getUserId(request);
+
+    if (!userId.isPresent()) {
       return;
     }
 
-    List<Group> groups = datastore.getAllGroups();
+    List<Group> groups = datastore.getNotJoinedGroups(userId.get());
```
Comment (nataliakrylova) [inline src/main/java/com/google/lecturechat/servlets/GroupsServlet.java:52]: Same here (it's not "all available groups" anymore).

Comment (nataliakrylova) [inline src/main/java/com/google/lecturechat/servlets/GroupsServlet.java:52]: Another question: do you think you'll ever need an opportunity to get all groups/events and not only the ones which user hasn't join?

File: src/main/webapp/events-script.js [hunk 4]
```diff
@@ -28,11 +28,14 @@ const eventsDictionary = {};
...
   /**
    * Creates a new event with the given parameters.
+   * @param {Long} id The id used to store the event in the database.
    * @param {String} title The title of the event.
    * @param {Date} start The start date of the event.
    * @param {Date} end The end date of the event.
    */
-  constructor(title, start, end) {
+  constructor(id, title, start, end) {
+    /** @private @const {LOng} */
```
Comment (nataliakrylova) [inline src/main/webapp/events-script.js:37]: (nit) long

File: src/main/webapp/events-script.js [hunk 5]
```diff
@@ -142,6 +148,56 @@ function addButtonToGetNewMonth(buttonClass, calendarHeader, date,
...
 
+/**
+ * Adds a button that can be used to open the chatroom of the event.
+ * @param {Element} eventOptionsElement The element that will include this
+ * button.
+ */
+function addChatroomButton(eventOptionsElement) {
+  const chatroomButton = createElement('button', 'rounded-button', 'Chatroom');
+  chatroomButton.addEventListener('click', function() {
+    // TODO: redirect the user to the chatroom
```
Comment (nataliakrylova) [inline src/main/webapp/events-script.js:159]: Do you have an issue for this? Maybe reference it somehow? TODO(issue #20): ... for example

File: src/main/webapp/events-script.js [hunk 6]
```diff
@@ -213,8 +272,24 @@ function displayEvents(date) {
   eventsContainer.innerHTML = '';
 
   for (let i = 0; i < events.length; i++) {
-    eventsContainer.appendChild(createEventElement(events[i]));
+    eventsContainer.appendChild(createEventElement(events[i], true));
```
Comment (nataliakrylova) [inline src/main/webapp/events-script.js:275]: Update documentation to explain that only the events user has joined will be displayed.

**Review by nataliakrylova — APPROVED** [review]

File: src/main/java/com/google/lecturechat/servlets/GroupsServlet.java (hunk 3 above)
Comment (AlexandraImbrisca) [reply src/main/java/com/google/lecturechat/servlets/GroupsServlet.java:52]: I think they are useful for testing the code, but I couldn't think of any particular scenario where we need all the groups/events and not only the ones which the user didn't join.

**Pull request merged**